agent/src/beerocks/monitor/monitor_thread.h

```diff
@@ -54,6 +54,9 @@ class monitor_thread : public beerocks::socket_thread {
     virtual std::string print_cmdu_types(const beerocks::message::sUdsHeader *cmdu_header) override;
 
 private:
+    bool handle_cmdu_control_ieee1905_1_message(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx);
```

Comment:
I noticed you escape this function if sd is nullptr. In this case, if you don't know how to work with a nullptr Socket, please _require_ it in your interface. The best would be:
```suggestion
    bool handle_cmdu_control_ieee1905_1_message(Socket &sd, ieee1905_1::CmduMessageRx &cmdu_rx);
```

Reply:
I agree, but note that this is a shortcoming in this entire file. That said, for new functions we can definitely do the right thing.